src/components/autofillFields/AutofillFields.tsx

```diff
@@ -1,20 +1,163 @@
+import { staticAutofillValues } from '@/utils/constants'
+import { TiptapEditorUtils } from '@/utils/tiptapEditorUtils'
+import { Editor } from '@tiptap/react'
+import { When } from '../hoc/When'
+import { useEffect, useState } from 'react'
+
 const AutofillFields = () => {
+  const appState = useAppState()
+
+  const tiptapEditorUtils = new TiptapEditorUtils(
+    appState?.appState.editor as Editor,
+  )
+
+  useEffect(() => {
+    if (!appState?.appState.selectedClient && !appState?.appState.readOnly)
+      return
+    appState?.setClientCompanyName('')
+    ;(async () => {
+      appState?.setLoading(true)
+      const res = await fetch(
+        `/api/companies?companyId=${appState?.appState.selectedClient?.companyId}`,
+      )
+      const { data } = await res.json()
+      if (data.name) {
+        appState?.setClientCompanyName(data.name)
+      } else {
+        appState?.setClientCompanyName('')
+      }
+      appState?.setLoading(false)
+    })()
+  }, [appState?.appState.selectedClient])
+
   return (
     <div className='p-5'>
       <p className='font-medium pb-5'>Autofill fields</p>
       <div className='flex flex-col gap-5'>
-        <AutofillText label={'client.givenName'} />
-        <AutofillText label={'client.familyName'} />
-        <AutofillText label={'client.email'} />
-        <AutofillText label={'client.company'} />
-        <AutofillText label={'client.address'} />
+        {/* readonly mode */}
+        <When condition={appState?.appState.readOnly as boolean}>
+          <AutofillTextReadonlyMode
+            label={`Given name: ${
+              appState?.appState.selectedClient?.givenName as string
+            }`}
+          />
+          <AutofillTextReadonlyMode
+            label={`Family name: ${
+              appState?.appState.selectedClient?.familyName as string
+            }`}
+          />
+          <AutofillTextReadonlyMode
+            label={`Email: ${
+              appState?.appState.selectedClient?.email as string
+            }`}
+          />
+          <AutofillTextReadonlyMode
+            label={`Company: ${
+              appState?.appState.selectedClientCompanyName
+                ? appState?.appState.selectedClientCompanyName
+                : ''
+            }`}
+          />
+          <AutofillTextReadonlyMode
+            label={`Address: ${
+              (appState?.appState.selectedClient?.address as string) || ''
+            }`}
+          />
+          {appState?.appState.selectedClient &&
+            Object.keys(
+              (appState?.appState.selectedClient as IClient)?.customFields,
+            ).length > 0 &&
+            Object.entries(
+              (appState?.appState.selectedClient as IClient)?.customFields,
+            ).map((value, key) => {
+              return (
+                <AutofillTextReadonlyMode
+                  key={key}
+                  label={`${value[0]}: ${value[1]}`}
+                />
+              )
+            })}
```

Comment:
I think this conditions can be reduces to 
```suggestion
          {
            Object.entries(
              (appState?.appState.selectedClient as IClient)?.customFields || {}
            ).map((value, key) => {
              return (
                <AutofillTextReadonlyMode
                  key={key}
                  label={`${value[0]}: ${value[1]}`}
                />
              )
            })}
```

Reply:
Thanks. This is working well! @saugatmaharjan